compiler/rustc_mir_transform/src/dataflow_const_prop.rs

```diff
@@ -573,7 +587,7 @@ impl<'a, 'tcx> Collector<'a, 'tcx> {
         map: &Map<'tcx>,
     ) -> Option<Const<'tcx>> {
         let ty = place.ty(self.local_decls, self.patch.tcx).ty;
-        let layout = ecx.layout_of(ty).ok()?;
+        let layout = ecx.layout_of(ty).discard_interp_error()?;
```

Comment:
I wonder if this should just call into `self.tcx.layout_of` -- having a `layout_of` query return an `InterpResult` that needs to be defused seems odd, imo.

Reply:
We could do that but then we have to pass the ParamEnv by hand. It doesn't mak the code shorter, overall.

There's a lot of code that has TyCtx + ParamEnv but sadly no commonly used type to pack the two leading to a lot of issues like this. :/